hail/python/hailtop/utils/utils.py

```diff
@@ -799,7 +799,7 @@ async def retry_transient_errors_with_debug_string(debug_string: str, warning_de
             if errors == 1 and is_retry_once_error(e):
                 return await f(*args, **kwargs)
             if not is_transient_error(e):
-                raise
+                raise ValueError(f"errors={errors}, delay={delay}") from e
```

Comment:
This smothers the real exception that the caller might expect, e.g. tests that test for a 404